doc/Clock commands/clock_managment_hld.md

```diff
@@ -0,0 +1,180 @@
+		* 3.2.2. [CLI](#Climodel)
+* 4. [Test Plan](#TestPlan)
+		* 4.1. [Unit Test cases](#UnitTestcases)
+
+
+### 1.1 <a name='Revision'></a>Revision
+
+|  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  0.1  | 01/2023 | Meir Renford	 | Phase 1 Design     |
+
+###  1.2. <a name='Scope'></a>Scope
+
+This document will address the high level design for NVOS clock commands:
+1.	Set/show date-time command
+2.	Set/show timezone command
+
+
+### 1.3 <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations 
+
+N/A
+
+### 1.4 Overview 
+
+The clock commands allow to set and review the current time parameters of the system - including: time, date and timezone. 
+
+### 1.5 Requirements
+####  1.5.1. <a name='FunctionalRequirements'></a>Functional requirements
+
+1. Any time configuration that will change in the system will change the system time. 
+
+####  1.5.2. <a name='ConfigurationManagementRequirements'></a>Configuration and Management Requirements
+The requirements from the module are: 
+1.	Set and show the system time and date.
+2.	Set and show the system timezone.
+
+
+##  2 <a name='Design'></a>Design
+
+###  2.1 <a name='High-LevelDesign'></a>High-Level Design
+
+The design of this feature is based on Linux command of <b>timedatectl</b>.<BR>
+(man page for timedatectl: https://man7.org/linux/man-pages/man1/timedatectl.1.html)
+
+All set operations are based on this command, executing directly in linux upon user CLI execution.
+The time/date/timezone are configured directly to Linux, and is persistent upon any reboot.
+
+System state of time/date/timezone will appear in 1 line as an output of existing "show clock" command.
+
+Set operations will be divided into 2 different commands:
+
+1. config clock set-timezone <timezone> (to set timezone command)
+	* will get single input as a string, and validate it is valid as part of ("timedatectl list-timezones" command).
+	Validation will be done either by YANG model, or (if not possible) by issueing relevant error to log.
+	* Value will be stored in db for future upgrade operations.
+	  Value is persistent upcon reboot.
+	* Linux timedatectl with set-timezone flag will be called.
+	  e.g. timedatectl set-timezone "Asia/Kolkata"
+	* <b>In case NTP is enabled -> timezone configuration is allowed and overrides the current time.</b>
+
+
+2. config clock set-date "<YYYY-MM-DD HH:MM:SS>" (to set time and date)
+	* Command will recieve single string with date-time format "<YYYY-MM-DD HH:MM:SS>"
+	* It will be possible to call command with the following options:
+		1.	Only with a date <YYYY-MM-DD>
+		2. 	Only with time <HH:MM:SS>
+		3.  both date and time "<YYYY-MM-DD HH:MM:SS>"
+	* Command does not need to be stored in DB.
+	* Linux timedatectl with set-time flag will be called.
+	  e.g. timedatectl set-time "2012-10-30 18:17:16"
+	* <b>In case NTP is enabled -> time/date set is NOT allowed and being blocked</b>
```

Comment:
as we store the timezone in config_db, how does image upgrades and config migration flows handles this config?

Reply:
Will timestamps & date time in various logs and cli outputs  be updated when timezone/date is modified? Could you please list the logs and cli commands in the HLD?

Reply:
> this

timezone configuration is part of DEVICE_METADATA, hence - configuration upgrade will be taken care of as any other sonic configuration.
will be taken care by hostcfgd.
added relevant comment to HLD (section 2).

Reply:
> Will timestamps & date time in various logs and cli outputs be updated when timezone/date is modified? Could you please list the logs and cli commands in the HLD?

Since system time will be changed, all commands that display current system time - will show current time (even if changed) accordingly.
Same goes with and log that takes system time on any new line print.